tests/0089-max_poll_interval.c

```diff
@@ -179,8 +195,19 @@ int main_0089_max_poll_interval(int argc, char **argv) {
         }
 
 
-        for (i = 0; i < 2; i++)
+        for (i = 0; i < 2; i++) {
                 rd_kafka_destroy_flags(c[i],
                                        RD_KAFKA_DESTROY_F_NO_CONSUMER_CLOSE);
+
+                if (use_log_queue)
+                        rd_kafka_queue_destroy(logq[i]);
+        }
+
+        SUB_TEST_PASS();
+}
+
+int main_0089_max_poll_interval(int argc, char **argv) {
+        do_test_with_optional_queue(rd_false);
+        do_test_with_optional_queue(rd_true);
```

Comment:
Done, also added a changelog entry

Reply:
I think if we're adding some new checks to an old test case, it's ok to change it as long as we don't change existing asserts.
This is also to avoid having tests that are too slow, especially integration tests.
It can also be ok with to parameterize some tests to avoid code duplication, as long as the tests are very similar and the resulting code is readable.

What do you think @pranavrth?
That said @milindl you don't have to revert this change you can leave them separate.

Reply:
I am okay with either option as per what we decide.
Both these tests are slow though and can't be run in make quick because of the timeout.

Reply:
I felt that we are touching multiple places in the old test so asked to separate.
Though anything is fine as far as the old test is not changed.